Build failure: firefox-{devedition, beta}-unwrapped

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

I tried to update my system to the latest version, but firefox-beta fails to build. Yes, somehow, it tries to download and install firefox-beta-unwrapped-135.0b9 instead, even if in my [config files](https://codeberg.org/BlastboomStrice/dotfiles/src/branch/main/.config/nixos-config/hosts/default/configuration.nix) I don't mention the unwrapped version anywhere..

The commands I used:
`sudo nix flake update --flake ~/.config/nixos-config/`
`sudo nixos-rebuild switch --flake ~/.config/nixos-config/`

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

https://hydra.nixos.org/build/291813476

### Relevant log output

Output of `nix log /nix/store/3msj81w89w7zlbq3hxmkckhbf86yc0jk-firefox-beta-unwrapped-135.0b9.drv`:

```console
Running phase: unpackPhase
Running phase: unpackPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/y5rnj7c5byk32i944k23sgrzcpkkvp7q-firefox-135.0b9.source.tar.xz
source root is firefox-135.0
setting SOURCE_DATE_EPOCH to timestamp 1737724845 of file "firefox-135.0/sourcestamp.txt"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/vxk7kwxf60bppmblksx8j70w3j7ggz4h-env_var_for_system_dir-ff133.patch
patching file toolkit/xre/nsXREDirProvider.cpp
applying patch /nix/store/6wsdh6246xbmlv3jr5q7qpl9ay3acq04-no-buildconfig-ffx121.patch
patching file docshell/base/nsAboutRedirector.cpp
patching file toolkit/content/jar.mn
Hunk #1 succeeded at 46 (offset 5 lines).
applying patch /nix/store/2f0970zfkxr14wb3iwq84dkwd7vh4f83-wasi-sdk-disable-reference-types.patch
patching file taskcluster/scripts/misc/wasi-sdk.patch
applying patch /nix/store/fqvq7azrbf7gqwr73gwd9hq11dmb8vm3-firefox-mac-missing-vector-header.patch
patching file security/sandbox/mac/Sandbox.h
patching script interpreter paths in mach build
mach: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python3"
build/build-clang/build-clang.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python3"
build/cargo-linker: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python3"
build/package/mac_osx/unpack-diskimage: interpreter directive changed from "#!/bin/bash" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash"
build/package/mac_osx/make-diskimage: interpreter directive changed from "#!/bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/checksums.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python"
build/clang-plugin/import_mozilla_checks.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bi>
build/pgo/profileserver.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python"
build/macosx/xcrun: interpreter directive changed from "#!/bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/qemu-wrap: interpreter directive changed from "#!/bin/bash" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash"
build/vs/generate_yaml.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/bin/python3"
build/unix/build-binutils/build-binutils.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash"
build/unix/build-hfsplus/build-hfsplus.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash"
build/unix/print-non-newline.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/unix/build-gcc/build-gcc.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash"
build/cargo-host-linker: interpreter directive changed from "#!/bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/autoconf/install-sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/autoconf/config.guess: interpreter directive changed from "#! /bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
build/autoconf/config.sub: interpreter directive changed from "#! /bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Configuring to generate profiling data
configure flags: --prefix=/nix/store/q8nakb43jy2qcj8dy1n4qhqxj00y0yyi-firefox-beta-unwrapped-135.0b9 --disable-tests --disable-updater --enable-application=browser -->
Creating global state directory from environment variable: /build/mozbuild
Running "pip check" to verify compatibility between the system Python and the "mach" site.
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check >
Creating local state directory: /build/mozbuild/srcdirs/firefox-135.0-42738f056eb7
Running "pip check" to verify compatibility between the system Python and the "build" site.
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check >
 0:02.85 Using Python 3.12.9 from /build/mozbuild/srcdirs/firefox-135.0-42738f056eb7/_virtualenvs/build/bin/python
 0:02.85 checking for vcs source checkout... no
 0:02.92 checking for a shell... /nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash
 0:02.93 checking for host system type... x86_64-unknown-linux-gnu
 0:02.94 checking for target system type... x86_64-unknown-linux-gnu
 0:03.30 checking whether cross compiling... no
 0:03.42 checking for the target C compiler... /nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang
 0:03.42 checking for ccache... not found
 0:03.50 checking whether the target C compiler can be used... yes
 0:03.50 checking for the target C++ compiler... /nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++
 0:03.59 checking whether the target C++ compiler can be used... yes
 0:03.70 checking for unwind.h... yes
 0:03.79 checking for target linker... lld
 0:04.01 checking for _Unwind_Backtrace... yes
 0:04.01 checking for Python 3... /build/mozbuild/srcdirs/firefox-135.0-42738f056eb7/_virtualenvs/build/bin/python (3.12.9)
 0:04.02 checking for wget... not found
 0:04.02 checking the target C compiler version... 19.1.7
 0:04.07 checking the target C compiler works... yes
 0:04.07 WARNING: The value of LD is not used by this build system.
 0:04.07 checking the target C++ compiler version... 19.1.7
 0:04.12 checking the target C++ compiler works... yes
 0:04.13 checking for the host C compiler... /nix/store/ls0g67nsklb2vn3vc9dnksa1adfgq32a-clang-wrapper-19.1.7/bin/cc
 0:04.22 checking whether the host C compiler can be used... yes
 0:04.22 checking the host C compiler version... 19.1.7
 0:04.27 checking the host C compiler works... yes
 0:04.27 checking for the host C++ compiler... /nix/store/ls0g67nsklb2vn3vc9dnksa1adfgq32a-clang-wrapper-19.1.7/bin/c++
 0:04.35 checking whether the host C++ compiler can be used... yes
 0:04.35 checking the host C++ compiler version... 19.1.7
 0:04.40 checking the host C++ compiler works... yes
 0:04.49 checking for host linker... lld
 0:04.54 checking for 64-bit OS... yes
 0:04.62 checking for new enough STL headers from libstdc++... yes
 0:04.83 checking for __thread keyword for TLS variables... yes
 0:04.83 checking for the assembler... /nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang
 0:04.87 checking for llvm-objdump... /nix/store/ppczfasmhhi2h0xfca89qm2rw6pdng9b-llvm-binutils-19.1.7/bin/llvm-objdump
 0:04.91 checking for ar... /nix/store/sg630gx6jqn1gqikvm9nkvaj0im1p5vh-llvm-binutils-wrapper-19.1.7/bin/ar
 0:04.98 checking whether ar supports response files... no
 0:05.02 checking for host_ar... /nix/store/ppczfasmhhi2h0xfca89qm2rw6pdng9b-llvm-binutils-19.1.7/bin/llvm-ar
 0:05.06 checking for nm... /nix/store/sg630gx6jqn1gqikvm9nkvaj0im1p5vh-llvm-binutils-wrapper-19.1.7/bin/nm
 0:05.11 checking for -mavxvnni support... yes
 0:05.16 checking for -mavx512bw support... yes
 0:05.21 checking for -mavx512vnni support... yes
 0:05.21 checking for pkg_config... /nix/store/601zbsyxn1zqx22yrif9yxa1x450qfc8-pkg-config-wrapper-0.29.2/bin/pkg-config
 0:05.22 checking for pkg-config version... 0.29.2
 0:05.22 checking whether pkg-config is pkgconf... no
 0:05.40 checking for strndup... yes
 0:05.60 checking for posix_memalign... yes
 0:05.77 checking for memalign... yes
 0:05.96 checking for malloc_usable_size... yes
 0:06.04 checking for malloc.h... yes
 0:06.10 checking whether malloc_usable_size definition can use const argument... no
 0:06.15 checking for stdint.h... yes
 0:06.21 checking for inttypes.h... yes
 0:06.26 checking for alloca.h... yes
 0:06.31 checking for sys/byteorder.h... no
 0:06.36 checking for getopt.h... yes
 0:06.43 checking for unistd.h... yes
 0:06.48 checking for nl_types.h... yes
 0:06.53 checking for cpuid.h... yes
 0:06.60 checking for fts.h... yes
 0:06.65 checking for sys/statvfs.h... yes
 0:06.71 checking for sys/statfs.h... yes
 0:06.76 checking for sys/vfs.h... yes
 0:06.85 checking for sys/mount.h... yes
 0:06.91 checking for sys/quota.h... yes
 0:06.98 checking for linux/quota.h... yes
 0:07.04 checking for linux/if_addr.h... yes
...skipping...
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_gamepad1.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/buil>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/fs/parent'
make[3]: Entering directory '/build/firefox-135.0/objdir/dom/geolocation'
mkdir -p '.deps/'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Geolocation.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/firefox-135>
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o GeolocationCoordinates.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/>
make[3]: Entering directory '/build/firefox-135.0/objdir/dom/grid'
mkdir -p '.deps/'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_grid0.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/gamepad'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o GeolocationPosition.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/fir>
make[3]: Entering directory '/build/firefox-135.0/objdir/dom/html/input'
mkdir -p '.deps/'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html_input0.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/b>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/grid'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o GeolocationPositionError.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/buil>
make[3]: Entering directory '/build/firefox-135.0/objdir/dom/html'
mkdir -p '.deps/'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html0.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/html/input'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html1.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
error: could not compile `gkrust` (lib)

Caused by:
  process didn't exit successfully: `/nix/store/4ranj0rglfp2ggycci9fmnkjc6z7zsd2-rustc-wrapper-1.84.1/bin/rustc --crate-name gkrust --edition=2015 toolkit/library/rus>
make[3]: *** [/build/firefox-135.0/config/makefiles/rust.mk:504: force-cargo-library-build] Error 101
make[3]: Leaving directory '/build/firefox-135.0/objdir/toolkit/library/rust'
make[2]: *** [/build/firefox-135.0/config/recurse.mk:72: toolkit/library/rust/target-objects] Error 2
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_geolocation0.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/>
v/include/nss -I/build/firefox-135.0/objdir/dist/include/nss -I/nix/store/xfsw9d5d1vi8891majlvd5nwix24w5m2-libpng-apng-1.6.46-dev/include/libpng16 -I/nix/store/fidbdi>
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html2.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/geolocation'
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html3.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html4.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
/nix/store/v7dvm6a4zsn3dbwcghrrc0vyq7asawmx-clang-wrapper-19.1.7/bin/clang++ -o Unified_cpp_dom_html5.o -c  -I/build/firefox-135.0/objdir/dist/stl_wrappers -I/build/f>
make[3]: Leaving directory '/build/firefox-135.0/objdir/dom/html'
make[2]: Leaving directory '/build/firefox-135.0/objdir'
make[1]: *** [/build/firefox-135.0/config/recurse.mk:34: compile] Error 2
make[1]: Leaving directory '/build/firefox-135.0/objdir'
make: *** [/build/firefox-135.0/config/rules.mk:359: default] Error 2
lines 3577-3616/3616 (END)
```

### Additional context

There are some hydra tests that fail, but are on different architectures:
https://hydra.nixos.org/build/291806033
https://hydra.nixos.org/build/291806025

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 6.12.17, NixOS, 25.05 (Warbler), 25.05.20250301.303bd80`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/7g9h6nlrx5h1lwqy4ghxvbhb7imm3vcb-source`

### Notify maintainers

@jopejoe1 ~~@ taku0 @ lovesegfault (wrong maintainers)~~
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/networking/browsers/firefox/packages/firefox-beta.nix
+++ b/pkgs/applications/networking/browsers/firefox/packages/firefox-beta.nix
@@ -10,11 +10,11 @@
 buildMozillaMach rec {
   pname = "firefox-beta";
   binaryName = pname;
-  version = "135.0b9";
+  version = "137.0b2";
   applicationName = "Firefox Beta";
   src = fetchurl {
     url = "mirror://mozilla/firefox/releases/${version}/source/firefox-${version}.source.tar.xz";
-    sha512 = "3007c2a8743e4444226e66f0c11f53f01255c09702deda7de83bbe134a19c324b8b49de78d3211b20bb82c7b2040127145d2e39ed8aa81c653ac4397c46476f6";
+    sha512 = "ac9d9635ede14506bf579735768cc643843c658122be89bb223f1c08702ee570757fbf091cf8191ad614688fff2330964ce728e51ea2519a801b838df94cc5d2";
   };
 
   meta = {

--- a/pkgs/applications/networking/browsers/firefox/packages/firefox-devedition.nix
+++ b/pkgs/applications/networking/browsers/firefox/packages/firefox-devedition.nix
@@ -10,13 +10,13 @@
 buildMozillaMach rec {
   pname = "firefox-devedition";
   binaryName = pname;
-  version = "135.0b9";
+  version = "137.0b2";
   applicationName = "Firefox Developer Edition";
   requireSigning = false;
   branding = "browser/branding/aurora";
   src = fetchurl {
     url = "mirror://mozilla/devedition/releases/${version}/source/firefox-${version}.source.tar.xz";
-    sha512 = "bf09f999b347492b841a26fcbcfb4d37e5f74528a05ffab47572dcaae01cb43e70baf58e83cc5153498a6a6ad2cb69507b628fba840090c91f4dbca62717a435";
+    sha512 = "a15eded14d6a4ddeb3798ac7bb72a9e8a6541a3ee76dc9051cf6865816079f8794e6defe2f9299f327976eefc1f4ea957d972b56d1f03816296b25f0c0173a12";
   };
 
   meta = {

